### DISCUSSION TITLE
Remove excessively repetitive/verbose service assignment in constructors

### DISCUSSION THREAD

**Discussion by audigex** [dotnet/csharplang#5216] [Language Ideas]
Currently when we inject a dependency, there's a somewhat repetitive, seemingly pointless amount of code involved

    private readonly ISomeService _someService;

    public MyClass(ISomeService someService)
    {
        _someService = someService;
    }

It feels like I write this (or something to it's effect) 200 times for every project

Is there any way we can introduce something like a `service` keyword, which can handle this for us and automatically assign the injected services? So instead of the above, we'd just have a single line like the following

    service ISomeService _someService;

**Comment by 333fred** [comment]
You're probably interested in either https://github.com/dotnet/csharplang/issues/4024 or https://github.com/dotnet/csharplang/issues/2691.

**Reply by TonyValenti** [reply]
I am happily waiting for required properties. Once we get that, I doubt I will never write another constructor again.￼￼￼

**Reply by 333fred** [reply]
I'll get there 😅

**Reply by marshalhayes** [reply]
Dang, lots of people beat me to this proposal. I've just starting using dependency injection with dotnet core and find this heavily verbose and repetitive as well. 

+1
